src/test/java/com/thebois/utils/MatrixUtilsTests.java

```diff
@@ -44,4 +62,96 @@ public void forEachElementGetsAllElementsInOrder(final Integer[][] matrix,
         assertThat(actualElements).containsExactlyElementsOf(expectedElements);
     }
 
+    @ParameterizedTest
+    @MethodSource("getMatricesToTest")
+    public void matrixToCollectionTest(final Optional<String>[][] matrix,
+                                       final Collection<String> expectedElements) {
+        // Arrange
+
+        // Act
+        final Collection<String> stringCollection = MatrixUtils.matrixToCollection(matrix);
+
+        // Assert
+        assertThat(stringCollection).isEqualTo(expectedElements);
+    }
+
+    private static Stream<Arguments> getCorrectCoordinatesToTest() {
+        final Optional<String>[][] stringOptionalMatrix = new Optional[][] {
+            { Optional.of("1"), Optional.empty(), Optional.of("3") },
+            { Optional.empty(), Optional.empty(), Optional.of("6") },
+            { Optional.of("7"), Optional.of("8"), Optional.of("9") },
+            };
+
+        return Stream.of(Arguments.of(stringOptionalMatrix, 0, 0, "1"),
+                         Arguments.of(stringOptionalMatrix, 2, 2, "9")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("getCorrectCoordinatesToTest")
+    public void testMatrixSpiralSearch(final Optional<String>[][] stringMatrix, final int startX,
+                                         final int startY,
+                                         final String expectedString) {
+
+        // Arrange
+
```

Comment:
Looks like you're arranging completely in the parameter method source, which is okay. You don't need to have empty comments like this.